### DISCUSSION TITLE
Connection is dropped for reader and publisher after 10 seconds

### DISCUSSION THREAD

**Discussion by brothercorvo** [bluenviron/mediamtx#831] [Questions]
## Which version are you using?

v0.0.0

## Which operating system are you using?

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [x] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

server disconnects publisher and reader after ~10 seconds

## Describe how to replicate the issue

1. start the server. API is activated
2. create a path using the API
3. Set the time out to 60 seconds
```
{
"readTimeout": "20s",
"writeTimeout": "20s"
}
```
4. publish with TAK ICU (https://drive.google.com/file/d/1Lz-b9S2kmdRF72UCAGZKAdS733v0_VLM/view?usp=sharing)
5. read with VLC
4. query REST API /v1/rtspsessions/list
5.  result:
```
{
"items": {
123374827: {
"remoteAddr": "XXX.42.255.101:40708",
"state": "publish"
},
648625851: {
"remoteAddr": "YYY174.157.179:26994",
"state": "read"
}
}
}
```
6. client start to load stream
7. before anything shows the connection is cut
8. query again the API  /v1/rtspsessions/list
9. result
```
{
"items": {}
}
```
yes / no

**Comment by brothercorvo** [comment]
the console message was:

```
2021/11/03 18:35:36 INF [1/0] [RTSP] [session 975494215] is publishing to path 'live/corvo', 1 track with UDP
2021/11/03 18:35:46 INF [0/0] [path live/corvo] destroyed
2021/11/03 18:35:46 INF [0/0] [RTSP] [session 975494215] closed (no UDP packets received in a while)
```
I managed to go around the issue by deleting all protocols but TCP

**Comment by aler9** [comment]
Hello, this is not a bug, the UDP transport protocol is not supported when there's a NAT/firewall in between. Please switch to the TCP transport protocol as described in the README.

[a comment by an automated account is not shown]
